client/rpc/src/main/kotlin/net/corda/client/rpc/internal/RPCClientProxyHandler.kt

```
@@ -444,12 +449,19 @@ object RpcClientObservableSerializer : Serializer<Observable<*>>() {
         }.dematerialize()
     }
 
+    private fun Input.readInvocationId() : InvocationId? {
+
+        val value = readString() ?: return null
```

The only invocation of the method shows (line 434) that returning null results in `IllegalStateException`. However the next line may throw different exception. Maybe instead of returning `null` throw this exception here. Also maybe catch exception from `readLong` and re-throw `IllegalStateException` to be consistent with exception case for readString()?

---

I'd rather keep this function pure and handle the problem directly. About `readLong()`, that returns a `long` primitive, so it would throw `NullPointerException` and there is no boxed equivalent, but id and timestamp are set together, so if the id is there there's no chance the timestamp will be missing.